content/blog/2021/04/2021-04-16-deprecated-ruby-runtime.adoc

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: "Deprecated ruby-runtime and associated plugins"
+tags:
+- ruby
+- plugins
+authors:
+- halkeye
+opengraph:
+  image: /images/post-images/2021-04-deprecating-ruby-runtime.png
+---
+Approximately
+https://github.com/jenkinsci/ruby-runtime-plugin/commit/d368b087fadb3282c9b537f0fa6d9a150b080c73[10 years ago, ruby-runtime was first released]. It was an experiment to let
+plugins to be written in ruby but still get integrated into the main
+java Jenkins process.
+
+Over the years though, the experiment has stopped being maintained, the
+dozen or so plugins that use it take a lot of user effort to make work,
+and the Jenkins ruby runtime does not support Java 11.
+Specifically the last few years have been really hard on the Jenkins
+core developers, as they still have to maintain all the hacks and
+processes to keep the ruby runtime barely working.
+
+One month from now, May 15th, 2021, the Jenkins update center will stop distributing
+ruby-runtime and its https://plugins.jenkins.io/ruby-runtime/#dependencies[associated plugins].
```

Comment:
```suggestion
ruby-runtime and its https://plugins.jenkins.io/ruby-runtime/#dependencies[associated plugins].
The Jenkins security team will also stop providing security fixes for these plugins.
```

Reply:
👎 We haven't done that anyway, see https://plugins.jenkins.io/gitlab-hook/

All of these plugins that we had findings for were abandoned, so https://www.jenkins.io/security/plugins/ applied.